### PR TITLE
Make utility function IsNumeric return true for numeric strings with trailing whitespace

### LINKED ISSUE #160: tixiGetDouble and tixiGetInteger fail, if the xpath element contains white space
Noticed in DLR-SC/tigl#710

### PR DESCRIPTION
**Pull request by joergbrech**
Since recently, we have the `isNumeric` check in `tixiGetDoubleElement` and `tixiGetIntegerElement` so that an error is returned for nonnumeric strings. However, `IsNumeric` also returned an error for strings that are numeric but contain trailing whitespaces. With this PR, trailing white spaces are ignored in the check.

Fixes #160
Addresses DLR-SC/tigl#710

### CODE REVIEW

File: src/tixiUtils.c [hunk 1]
```diff
@@ -410,11 +410,37 @@ char *substring(const char *str, int start_pos, int end_pos)
...
       return 0;
     }
 
-    strtod (s, &p);
+    strtod (s_trimmed, &p);
     return *p == '\0';
 }
+
+char* trim_trailing_whitespace(const char *str)
+{
```
Comment (rainman110) [inline src/tixiUtils.c:425]: We actually dont need to realloc. We just have to set the nul terminator:
https://stackoverflow.com/a/1431206

Comment (joergbrech) [reply src/tixiUtils.c:425]: But would that also work on `const char *str`?

Comment (rainman110) [reply src/tixiUtils.c:425]: There's probably a way to make it work with `const char*`

Comment (rainman110) [reply src/tixiUtils.c:425]: const char * means "Pointer to const char". I.e. you can't change the characters, but you can change the pointer. If you do not want to change the pointer, you need `char const * const`.

Comment (joergbrech) [reply src/tixiUtils.c:425]: I will change it to work in place with `char *`. Thats enough for our usecase.

**Pull request merged**